File: container-with-the-most-water/solve.cpp
class Solution {
public:
    int max(int first, int second){
            
             if(first >= second){
                 return first;
             }
             return second;
    }
    
    int min(int first, int second){
            
             if(first <= second){
                 return first;
             }
             return second;
    }
    
    int maxArea(vector<int>& height) {
      
        int maxArea = 0;
        int a=0;
        int b=0;
        
        for(int i=0; i <height.size()-1; i++){
            maxArea = max(maxArea,(height.size()-a-b-1)*min(height.at(a),height.at(height.size()-1-b)));
            if(height.at(a)<=height.at(height.size()-1-b)){
                a++;
            }else{
                b++;
            }
            
        }
        return maxArea;
    }
};

File: min-cost-climbing-stairs/solve.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        n = len(cost)
        if n <= 1 :
            return 0

        dp = [0] * (n + 1)

        for i in range(2 , n+ 1) :
            dp[i] = min(cost[i - 1] + dp[i - 1] , cost[i - 2] + dp[i - 2])
        
        return dp[n]
